### PR TITLE
Allow to download an artifact from the UI (#786)

### PR DESCRIPTION
**Pull request by AnbazhakanSubramaniam**
This feature allows to download the artifact from the Artifact details
table

Signed-off-by: usb1cob <Anbazhakan.Subramaniam@in.bosch.com>

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Comment by schabdo** [comment]
@laverman could you please have a look and support @AnbazhakanSubramaniam in his contribution? Thanks!

**Comment by schabdo** [comment]
@laverman ping

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/details/ArtifactDetailsLayout.java [hunk 1]
```diff
@@ -290,19 +296,71 @@ private void addGeneratedColumnButton(final Table table) {
...
+        fileDownloader.setErrorHandler(new ErrorHandler() {
+
+            /**
+             * Error handler for file downloader
+             */
+            private static final long serialVersionUID = 4030230501114422570L;
+
+            @Override
+            public void error(com.vaadin.server.ErrorEvent event) {
+                uINotification.displayValidationError(i18n.getMessage("message.artifact.download.failure"));
```
Comment (laverman) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/details/ArtifactDetailsLayout.java]: Would you mind adding a variable in UIMessageIdProvider.java for that?

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/utils/UIMessageIdProvider.java [hunk 2]
```diff
@@ -15,6 +15,8 @@
  */
 public final class UIMessageIdProvider {
 
+    public static final String ARTIFACT_DOWNLOAD = "artifact.download";
```
Comment (laverman) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/utils/UIMessageIdProvider.java]: Please add a respective message to the `messages.properties` file (as you did previously: `message.artifact.download.failure = Artifact Download Failed` )

**Comment by schabdo** [comment]
@laverman if it looks good to you can you please accept the PR and merge it?

**Pull request merged**